lib/std/fmt/parse_float.zig

```diff
@@ -415,14 +47,105 @@ test "fmt.parseFloat" {
+    try testing.expectEqual(try parseFloat(f32, "0x0.ffffffp128"), 0x0.ffffffp128);
+    try testing.expectEqual(try parseFloat(f32, "0x0.1234570p-125"), 0x0.1234570p-125);
+    // Max normalized value.
+    try testing.expectEqual(try parseFloat(f32, "0x1.fffffeP+127"), math.floatMax(f32));
+    try testing.expectEqual(try parseFloat(f32, "-0x1.fffffeP+127"), -math.floatMax(f32));
+    // Min normalized value.
+    try testing.expectEqual(try parseFloat(f32, "0x1p-126"), math.floatMin(f32));
+    try testing.expectEqual(try parseFloat(f32, "-0x1p-126"), -math.floatMin(f32));
+    // Min denormal value.
+    try testing.expectEqual(try parseFloat(f32, "0x1P-149"), math.floatTrueMin(f32));
+    try testing.expectEqual(try parseFloat(f32, "-0x1P-149"), -math.floatTrueMin(f32));
+}
+
+test "fmt.parseFloat hex.f64" {
+    try testing.expectEqual(try parseFloat(f64, "0x1p0"), 1.0);
+    try testing.expectEqual(try parseFloat(f64, "-0x1p-1"), -0.5);
+    try testing.expectEqual(try parseFloat(f64, "0x10p+10"), 16384.0);
+    try testing.expectEqual(try parseFloat(f64, "0x10p-10"), 0.015625);
+    // Max normalized value.
+    try testing.expectEqual(try parseFloat(f64, "0x1.fffffffffffffp+1023"), math.floatMax(f64));
+    try testing.expectEqual(try parseFloat(f64, "-0x1.fffffffffffffp1023"), -math.floatMax(f64));
+    // Min normalized value.
+    try testing.expectEqual(try parseFloat(f64, "0x1p-1022"), math.floatMin(f64));
+    try testing.expectEqual(try parseFloat(f64, "-0x1p-1022"), -math.floatMin(f64));
+    // Min denormalized value.
+    //try testing.expectEqual(try parseFloat(f64, "0x1p-1074"), math.floatTrueMin(f64));
+    try testing.expectEqual(try parseFloat(f64, "-0x1p-1074"), -math.floatTrueMin(f64));
+}
+test "fmt.parseFloat hex.f128" {
+    try testing.expectEqual(try parseFloat(f128, "0x1p0"), 1.0);
+    try testing.expectEqual(try parseFloat(f128, "-0x1p-1"), -0.5);
+    try testing.expectEqual(try parseFloat(f128, "0x10p+10"), 16384.0);
+    try testing.expectEqual(try parseFloat(f128, "0x10p-10"), 0.015625);
+    // Max normalized value.
+    try testing.expectEqual(try parseFloat(f128, "0xf.fffffffffffffffffffffffffff8p+16380"), math.floatMax(f128));
+    try testing.expectEqual(try parseFloat(f128, "-0xf.fffffffffffffffffffffffffff8p+16380"), -math.floatMax(f128));
+    // Min normalized value.
+    try testing.expectEqual(try parseFloat(f128, "0x1p-16382"), math.floatMin(f128));
+    try testing.expectEqual(try parseFloat(f128, "-0x1p-16382"), -math.floatMin(f128));
+    // // Min denormalized value.
+    try testing.expectEqual(try parseFloat(f128, "0x1p-16494"), math.floatTrueMin(f128));
+    try testing.expectEqual(try parseFloat(f128, "-0x1p-16494"), -math.floatTrueMin(f128));
+
+    // NOTE: We are performing round-to-even. Previous behavior was round-up.
+    // try testing.expectEqual(try parseFloat(f128, "0x1.edcb34a235253948765432134674fp-1"), 0x1.edcb34a235253948765432134674fp-1);
```

Comment:
This is a specific example of where I presume the rounding difference is being a factor since we are off by one here. Haven't done a close verification but once stage2 is the base compiler and we have consistent rounding across all implementations these errors should cease.